src/topology/algebra/ring.lean

```diff
@@ -103,6 +107,10 @@ instance prod_ring {β : Type*}
   [ring β] [topological_space β] [topological_ring β] : topological_ring (α × β) :=
```

Comment:
It would probably make sense to rename this instance to `prod.topological_ring` to match. The current name is just bad.

In fact, I think you can remove the instance names both from the `prod` instances and the new `pi` instances, and Lean will pick the right names for you.

Reply:
I used `pi.topological_(semi)ring` to be consistent with `pi.topological_group` in `topology/algebra/group`, but I can remove the names if you think that's better.

Reply:
Note that lean will generate exactly that name anyway, so that doesn't really matter - feel free to leave the names explicit. Changing the prod name is the bit that matters

Reply:
I deleted the names and edited the docstring.